pkg/services/featuremgmt/models.go

```diff
@@ -2,11 +2,21 @@ package featuremgmt
 
 import (
 	"bytes"
+	"context"
 	"encoding/json"
 )
 
 type FeatureToggles interface {
-	IsEnabled(flag string) bool
+	// Check if a feature is enabled for a given request.
```

Comment:
```suggestion
	// Check if a feature is enabled for a given context.
```

background jobs etc.